lms/static/scripts/frontend_apps/components/test/GradingControls-test.js

```diff
@@ -271,6 +263,49 @@ describe('GradingControls', () => {
     );
   });
 
+  context('when has unsaved grade changes', () => {
+    const setUnsavedChanges = wrapper =>
+      act(() => {
+        wrapper.find('SubmitGradeForm').props().onUnsavedChanges(true);
+      });
+    const getSelectedStudent = wrapper =>
+      wrapper.find('StudentSelector').props().selectedStudent;
+
+    [
+      // Selected student should still be the first
+      {
+        discardChanges: false,
+        expectedStudentAfterChange: () => fakeStudents[0],
+      },
+
+      // Selected student should change from first to second
+      {
+        discardChanges: true,
+        expectedStudentAfterChange: () => fakeStudents[1],
+      },
+    ].forEach(({ discardChanges, expectedStudentAfterChange }) => {
+      it('"cancels" selected user change if instructor selected to continue editing', async () => {
+        const wrapper = renderGrader();
+        await selectFirstStudent(wrapper);
+
+        await setUnsavedChanges(wrapper);
+        wrapper.update();
+        fakeConfirm.resolves(discardChanges);
+
+        // Try to select another student
+        await act(() => {
+          wrapper
+            .find('StudentSelector')
+            .props()
+            .onSelectStudent(fakeStudents[1]);
+        });
```

Comment:
Same issue as in the above comment about `act`. This works, but I suspect is brittle in that if async activity in `onSelectStudent` took more than one microtask, the code would break.